### ISSUE TITLE
Connection::run(query: Insert) failed to unwrap lastInsertRowid with OnConflict.Ignore

### ISSUE THREAD

**Issue by asfdfdfd** [stephencelis/SQLite.swift#235]
branch: swift-2
commit: 8bd27bcd5d0e6461620e51a651156d57c1c2c6cb

Model

``` swift
class News {

    let id: Int
    let header: String

    init(row: Row) {
        id = row[NewsColumn.Id]
        header = row[NewsColumn.Header]
    }
}
```

Insert code

``` swift
try db.run(tableNews.insert(or: OnConflict.Ignore,
    NewsColumn.Id <- (rawNews["id"] as! Int),
    NewsColumn.Header <- (rawNews["name"] as! String),
))
```

This insert code will fail [here](https://github.com/stephencelis/SQLite.swift/blob/swift-2/Source/Typed/Query.swift#L951) if model with the same Id already exists.

**Comment by asfdfdfd** [comment]
https://www.sqlite.org/c3ref/last_insert_rowid.html

The sqlite3_last_insert_rowid(D) interface returns the rowid of the most recent successful INSERT into a rowid table [...]

An INSERT that fails due to a constraint violation is not a successful INSERT and does not change the value returned by this routine. Thus [...], INSERT OR IGNORE, [...] make no changes to the return value of this routine when their insertion fails.

**Comment by ahusseini** [comment]
This method is the culprit, `self.lastInsertRowid` is nullable, but the return type is Int64.
```swift
public func run(query: Insert) throws -> Int64 {
        let expression = query.expression
        return try sync {
            try self.run(expression.template, expression.bindings)
            return self.lastInsertRowid!
        }
    }
```

Temporary fix:
```swift
public func run(query: Insert) throws -> Int64 {
        let expression = query.expression
        return try sync {
            try self.run(expression.template, expression.bindings)
            let returnId = self.lastInsertRowid ?? -1
            return returnId
        }
    }
```

**Comment by asfdfdfd** [comment]
Also [here is](https://github.com/asfdfdfd/SQLite.swift/commit/a993a74b1574007d4cedb9e27676d20ded8f58a2) alternative fix, but it was rejected.

**Comment by ahusseini** [comment]
I tried that fix, but it does not build for me.

**Comment by asfdfdfd** [comment]
With my fix you have to check return value on nil. With 'if let' pattern, for example.

**Comment by ahusseini** [comment]
True, also I'm not sure how Stephen feels about returning -1 for failed Insert statements. We shall see.

**Comment by iNoles** [comment]
Since a row id should be unique number with Ignore, any column wouldn't fix a problem. I think insert methods should just do db.run without lastInsertRowid.

**Comment by iNoles** [comment]
@stephencelis How about
```
public func run(query: Insert) throws -> Connection {
        let expression = query.expression
        return try sync {
            try self.run(expression.template, expression.bindings)
            return self
        }
    }
```

**Comment by jberkel** [comment]
Fixed in 75394dab16991f64672dbf8a11fbb8e80c17a2e7
